tests/integration/collection_description/updates/add/collections_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package add
+
+import (
+	"testing"
+
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+)
+
+func TestColDescrUpdateAddCollections_WithUndefinedID_Errors(t *testing.T) {
+	test := testUtils.TestCase{
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Users {}
+				`,
+			},
+			testUtils.PatchCollection{
+				Patch: `
+					[
+						{ "op": "add", "path": "/2", "value": {"Name": "Dogs"} }
+					]
+				`,
+				ExpectedError: "collection ID cannot be zero",
```

Comment:
Would add another test where it is explicitly `0`:
`{ "op": "add", "path": "/2", "value": {"ID": 0, "Name": "Dogs"} }`

Reply:
Will add

- [ ] Add explicit zero add test